### PR TITLE
Revamp the `Asynchronous logging` page

### LINKED ISSUE #2534: Revamp the `Asynchronous logging` page
As agreed in [the site structure document](https://docs.google.com/document/d/10Fu7oqDzdM_D6LbexzwX9arh51Tic7AGvkWTQrL6jjQ/edit), revamp the `Asynchronous logging` page such that

- Revamp the existing content in `async.adoc`
  - Address ` YieldingWaitStrategy` issue shared in https://github.com/apache/logging-log4j2/issues/2169#issuecomment-2094011204
- Massage intro and trade-offs to accommodate the impatient
- Remove performance figures
- Either remove or push to a far corner the under-the-hood details
- Target `2.x`, port to `main`

### PR DESCRIPTION
**Pull request by ppkarwasz**
We refresh the asynchronous logging page by:
    
 * marking more clearly that _asynchronous logging_ is not the solution to all performance problems. Depending on the case it could give all disadvantage of async processing without any measurable advantages.
  * removes the system properties and points to the appropriate "Configuration properties" section.
  * refreshes the configuration examples by adding an example for each of the configuration formats.
    
 Closes #2534.

### CODE REVIEW

File: src/site/antora/modules/ROOT/pages/manual/async.adoc [hunk 1]
```diff
@@ -14,525 +14,375 @@
...
++
+* implement an
+link:../javadoc/log4j-core/org/apache/logging/log4j/core/async/RingBufferLogEvent.html[ExceptionHandler<? super RingBufferLogEvent>]
+* set its fully qualified class name as value of the
+xref:manual/systemproperties.adoc#log4j2.asyncLoggerExceptionHandler[log4j2.asyncLoggerExceptionHandler]
+configuration property.
+
+Mixed synchronous/asynchronous::
++
+If you use a mix of sync and async loggers you need to:
```
Comment (vy) [inline src/site/antora/modules/ROOT/pages/manual/async.adoc:363]:
```suggestion
If you use a mix of synchronous and asynchronous loggers you need to:
```

File: src/site/antora/modules/ROOT/pages/manual/async.adoc [hunk 2]
```diff
@@ -14,525 +14,375 @@
...
++
+* implement a
+link:../javadoc/log4j-core/org/apache/logging/log4j/core/async/AsyncLoggerConfigDisruptor.Log4jEventWrapper.html[ExceptionHandler<? super AsyncLoggerConfigDisruptor.Log4jEventWrapper>]
+* set its fully qualified class name as value of the
+xref:manual/systemproperties.adoc#log4j2.asyncLoggerConfigExceptionHandler[log4j2.asyncLoggerConfigExceptionHandler]
+configuration property.
 
 [#UnderTheHood]
```
Comment (vy) [inline src/site/antora/modules/ROOT/pages/manual/async.adoc:371]: I'd just remove this section. We already do Disruptor marketing at the top and several other places where users need to plug in Disruptor components.

**Review by vy — CHANGES_REQUESTED** [review]

File: src/site/antora/modules/ROOT/pages/manual/async.adoc [hunk 3]
```diff
@@ -14,525 +14,375 @@
...
+[IMPORTANT]
+====
+When using an asynchronous logger context you should use only `Root` and `Logger` elements (cf.
+xref:manual/configuration.adoc#configuring-loggers[Logger configuration]).
+
+If you use `AsyncRoot` and `AsyncLogger` configuration elements, two asynchronous barriers will be created instead of one, which will impair performance.
+====
 
 [[SysPropsAllAsync]]
 === Configuration Properties
```
[the remark this answers is not in the document]
Comment (ppkarwasz) [reply src/site/antora/modules/ROOT/pages/manual/async.adoc:187]: At first I thought that the properties bloat the page, but maybe you are right: let's include them.

File: src/site/antora/modules/ROOT/pages/manual/async.adoc [hunk 4]
```diff
@@ -14,525 +14,375 @@
...
+If your application is running in an environment where CPU resources are scarce, like a VM with a single vCPU, starting another thread is not likely to give better performance.
 
-[#AllAsync]
-== Making All Loggers Asynchronous
 
-NOTE: _Log4j-2.9 and higher require disruptor-3.3.4.jar or higher on the
-classpath. Prior to Log4j-2.9, disruptor-3.0.0.jar or higher was
-required._
+[id=asynchronous-loggers]
+== Asynchronous loggers
```
[the remark this answers is not in the document]
Comment (ppkarwasz) [reply src/site/antora/modules/ROOT/pages/manual/async.adoc]: I chose 1.

File: src/site/antora/modules/ROOT/pages/manual/async.adoc [hunk 5]
```diff
@@ -14,525 +14,375 @@
...
+You can choose between:
 
-A configuration that does not require location might look like:
+** <<AllAsync>>, which gives a better performance,
+** <<MixedSync-Async>>, which gives more flexibility.
 
-[source,xml]
+* *Asynchronous appenders* have been enhanced to flush to the OS at the end of a batch, when the queue becomes empty.
++
+This produces the same result as the xref:manual/appenders.adoc#immediateFlush[immediateFlush] attribute of many appenders: it guarantees that the batch of log events is passed to the OS, but does not guarantee that the OS forwards the data to the appropriate physical device.
```
[the remark this answers is not in the document]
Comment (ppkarwasz) [reply src/site/antora/modules/ROOT/pages/manual/async.adoc]: I moved the note to `appenders.adoc`.

File: src/site/antora/modules/ROOT/pages/manual/async.adoc (hunk 3 above)
[the remark this answers is not in the document]
Comment (ppkarwasz) [reply src/site/antora/modules/ROOT/pages/manual/async.adoc]: Done

File: src/site/antora/modules/ROOT/pages/manual/async.adoc [hunk 6]
```diff
@@ -14,525 +14,375 @@
...
+----
+====
+
+<1> All the appenders referenced by `Root` and `Logger` are called synchronously.
+This is especially important for audit logging, since exceptions can be forwarded to the caller.
+<2> All the appenders references by `AsyncRoot` and `AsyncLogger` are called asynchronously.
+These log statements will cause a smaller latency for the caller.
 
 [[SysPropsMixedSync-Async]]
 === Configuration Properties
```
[the remark this answers is not in the document]
Comment (ppkarwasz) [reply src/site/antora/modules/ROOT/pages/manual/async.adoc]: Done

File: src/site/antora/modules/ROOT/pages/manual/async.adoc (hunk 2 above)
Comment (ppkarwasz) [reply src/site/antora/modules/ROOT/pages/manual/async.adoc]: Fixed in 13aea96008fe4d6757842f8a090b39f69c827384

File: src/site/antora/modules/ROOT/pages/manual/async.adoc [hunk 7]
```diff
@@ -14,525 +14,375 @@
...
+include::example$manual/configuration/custom-wait-strategy.properties[lines=17..-1]
 ----
+====
+
+[#Location]
+== Location
 
-The specified class must also have a public no-argument constructor;
-Log4j will instantiate an instance of the specified factory class and use this factory to create the WaitStrategy used by all Async Loggers.
+Computing the location of a log event is an expensive operation.
```
[the remark this answers is not in the document]
Comment (ppkarwasz) [reply src/site/antora/modules/ROOT/pages/manual/async.adoc]: Fixed in a7854e8f786d001e38f28c2121f7428e9ab47af8

File: src/site/antora/modules/ROOT/pages/manual/async.adoc [hunk 8]
```diff
@@ -14,525 +14,375 @@
...
+====
+
+[#AllAsync]
+== Making all loggers asynchronous
+
+This is simplest to configure and gives the best performance.
+
+To make all logger asynchronous, you need to set the
+xref:manual/systemproperties.adoc#log4j2.contextSelector[log4j2.contextSelector]
+property to one of the asynchronous logger context selectors:
```
[the remark this answers is not in the document]
Comment (ppkarwasz) [reply src/site/antora/modules/ROOT/pages/manual/async.adoc:89]: Fixed in a7854e8f786d001e38f28c2121f7428e9ab47af8 by ending the section here.

**Review by vy — APPROVED** [review]

**Pull request merged**